SPIE2016-9913-16/9913-16.tex

```diff
@@ -0,0 +1,303 @@
+For heavily used public APIs it would seem to be clear that arguments should be quantities wherever possible.
+The question for LSST is how far down in the stack quantities should be used given the performance hit associated with quantities over plain variables.
+Additionally, how much more complicated does the C++/python interface become if SWIG has to be taught how to convert quantities to the correct form before passing the value to the C++.
+This is especially critical for large arrays, supplied externally, being passed through where we do not want to do a copy operation before entering C++ and also do not want to normalize units for millions of data points at that time.
+Large data arrays and, in particular, table columns returned from C++ layers would benefit from being returned as quantities, as this gives much more flexibility to users of the data and minimizes the risk of misinterpretation of values if units are later changed for a column or the user relies entirely on documentation to interpret values.
+The proposal is that LSST software adopts quantities on a case-by-case basis and are continuously on the look out for  any performance impact associated with quantities.
+
+\section{Proposal}
+
+The concrete proposals from this investigation are therefore:
+
+\begin{itemize}
+\item Change EUPS packaging to use a \texttt{setup.py} and upload it to PyPI.
+\item Investigate the addition of a \texttt{setup.py} for standard LSST Python packages in addition to our current Scons-based build system.
+\item Support export of LSST tables to Astropy tables.
+\item Determine whether STC-2 is a suitable basis for WCS interoperability.
+\item Remove LSST astronomical coordinate classes and replace with Astropy coordinates in Python and a generic spherical coordinates class in C++.
+\item Investigate migration away from SWIG for C++ bindings.
+\item Contribute the LSST \emph{footprint} code to Astropy.
+\item Create an Astropy affiliated package from the AFW detection package.
+\item Investigate rationalizing of LSST and Astropy spherical geometry region handling.
+\item Modify \texttt{Exposure} to implement an \texttt{NDData}-like interface.
+\item Work with the Astropy project to add required functionality such as pixel origins to \texttt{NDData}.
+\item LSST will allow Astropy quantity objects to be used in public interfaces on a case by case basis.
+
+\end{itemize}
+
+With these changes implemented LSST will have a solid base from which to assess whether it is feasible to migrate more LSST code to the community as affiliated packages.
+It is probable that LSST will start to migrate the existing codebase entirely to Astropy classes where appropriate classes exist, and in particular adopt \texttt{astropy.time} and \texttt{astropy.coordinates} as soon as possible.
```

Comment:
We haven't mentioned `astropy.time` before this, right?  It's not clear to me we can "just adopt" these, given that we need date/time objects in `Calib` and `Calib` has to go in `Exposure`.  If you've done the work to demonstrate that this is possible, we should either discuss it in this paper or leave `astropy.time` out entirely.